### PR TITLE
refactor: grpc and oauth2service refresh token

### PR DESCRIPTION
**Pull request by eapg**
* deleting redundant call to token validation
* refactoring grpc oauth2 refresh token exception to PERMISSION_DENIED.

### CODE REVIEW

File: src/main/java/com/ea/restaurant/grpc/GrpcMongoOrderStatusHistoryServiceImpl.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public void insertMongoOrderStatusHistoriesFromPythonEtl(
               .build());
     } catch (Exception e) {
       responseObserver.onError(
-          Status.INTERNAL.withDescription(e.getMessage()).withCause(e).asRuntimeException());
+          Status.PERMISSION_DENIED.withDescription(e.getMessage()).withCause(e).asRuntimeException());
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/grpc/GrpcMongoOrderStatusHistoryServiceImpl.java:43]: This should be an internal error. Why are you sending a `PERMISSION_DENIED` error if something happens within `insertMongoOrderStatusHistoriesFromPythonEtl`?

I don't think any of this operations will generate a `PERMISSION_DENIED` exception.

File: src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java [hunk 2]
```diff
@@ -86,7 +86,6 @@ public Oauth2TokenResponseDto refreshToken(
       throws BadJOSEException, ParseException, JOSEException {
     validateClientCredentials(clientId, clientSecret);
     try {
-      Oauth2Util.validateToken(accessToken, oauth2SecretKey);
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java:89]: why are you removing this?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/main/java/com/ea/restaurant/grpc/GrpcMongoOrderStatusHistoryServiceImpl.java (hunk 1 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/grpc/GrpcMongoOrderStatusHistoryServiceImpl.java]: I was a mistake. i fixed.

File: src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java (hunk 2 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java:89]: because the method getTokenDecode() bellow call validateToken().

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**